### PR TITLE
Removals of EntityList, EntitySequence, JobManager, and Controller

### PR DESCRIPTION
**Pull request by juliaputko**

### CODE REVIEW

File: smartsim/_core/control/job.py
Comment (amandarichardsonn) [inline smartsim/_core/control/job.py:1]: On line 194 of job.py -> The Job docstring seems out of date.

File: smartsim/_core/control/manifest.py [hunk 1]
```diff
@@ -127,24 +112,16 @@ def _check_names(deployables: t.List[t.Any]) -> None:
...
-        for entity_list in self.all_entity_lists:
-            if len(entity_list) < 1:
-                raise ValueError(f"{entity_list.name} is empty. Nothing to launch.")
-
     def __str__(self) -> str:
         output = ""
         e_header = "=== Ensembles ===\n"
-        m_header = "=== Applications ===\n"
-        db_header = "=== Feature Stores ===\n"
+        a_header = "=== Applications ===\n"
```
Comment (amandarichardsonn) [inline smartsim/_core/control/manifest.py:123]: nice catch

File: smartsim/_core/control/job.py [hunk 2]
```diff
@@ -200,17 +201,15 @@ def __init__(
...
-        entity: t.Union[SmartSimEntity, EntitySequence[SmartSimEntity], JobEntity],
+        entity: t.Union[SmartSimEntity, JobEntity],
         launcher: str,
-        is_task: bool,
     ) -> None:
         """Initialize a Job.
 
         :param job_name: Name of the job step
         :param job_id: The id associated with the job
         :param entity: The SmartSim entity(list) associated with the job
```
Comment (amandarichardsonn) [inline smartsim/_core/control/job.py:211]: I am not sure if entity(list) still applies?

File: smartsim/_core/control/manifest.py [hunk 3]
```diff
@@ -127,24 +112,16 @@ def _check_names(deployables: t.List[t.Any]) -> None:
     @staticmethod
     def _check_types(deployables: t.List[t.Any]) -> None:
         for deployable in deployables:
-            if not isinstance(deployable, (SmartSimEntity, EntitySequence)):
+            if not isinstance(deployable, (SmartSimEntity)):
```
Comment (amandarichardsonn) [inline smartsim/_core/control/manifest.py:115]: You do not need the extra () around SmartSimEntity

**Review by amandarichardsonn — APPROVED** [review]
LGTM!

Could you go into pyproject.toml and delete the files that you deleted from the list starting at line 150

File: smartsim/_core/control/manifest.py (hunk 3 above)
Comment (juliaputko) [reply smartsim/_core/control/manifest.py:115]: changed

File: smartsim/_core/control/job.py (hunk 2 above)
Comment (juliaputko) [reply smartsim/_core/control/job.py:211]: removed

File: smartsim/_core/control/job.py
Comment (juliaputko) [reply smartsim/_core/control/job.py:1]: removed references to controller

**Review by mellis13 — APPROVED** [review]
LGTM!

**Pull request merged**